odl/discr/grid.py

```diff
@@ -681,9 +684,46 @@ def append(self, other):
         See Also
         --------
         insert
+        extend
         """
         return self.insert(self.ndim, other)
 
+    def extend(self, seq):
+        """Return a copy of ``self``, extended by all members of ``seq``.
+
+        Parameters
+        ----------
+        seq : sequence of `RectGrid`
```

Comment:
~~Well, to some degree it's already too late since we're already down that route with `insert` and `append`. For sure `RectGrid` and `RectPartition` are no container types, so there's a bit of tension with the container association.
The only way to solve this would be to allow these types to be constructed with one or more of their own type, e.g.,~~
```python
>>> grid = RectGrid(grid1, grid2, ..., gridN)
```
...
... 
Oh, this is really bad:
```python
>>> g = RectGrid([0, 1, 2], [0, 1, 4])
>>> g.extend(g)
RectGrid(
    [0.0, 1.0, 2.0],
    [0.0, 1.0, 4.0],
    [0.0],
    [0.0, 1.0, 4.0],
    [1.0],
    [0.0, 1.0, 4.0],
    [2.0],
    [0.0, 1.0, 4.0]
)
```
Clear oops, so I think `extend has to go away again.